tests/pipelines/test_train_step.py

```diff
@@ -1,6 +1,8 @@
 import os
```

Comment:
After https://github.com/mlflow/mlflow/pull/6823 is merged, can we add test coverage for `parallelism > 1`?